File: LeetCodeProblemsCpp/LeetCodeProblemsCpp/Problems/LeetCode3477FruitsIntoBasketsII.cpp
#include "LeetCode3477FruitsIntoBasketsII.h"
using namespace std;

int LeetCode3477FruitsIntoBasketsII::numOfUnplacedFruits(vector<int>& fruits, vector<int>& baskets)
{
    int fruitsPlaced{0};
    for (auto fruit : fruits)
    {
        for (int i = 0, e = static_cast<int>(baskets.size()); i < e; i++)
        {
            if (baskets[i] < fruit)
                continue;
            baskets[i] = -1;
            fruitsPlaced++;
            break;
        }
    }
    return fruits.size() - fruitsPlaced;
}

File: LeetCodeProblemsGo/LeetCode3477FruitsIntoBasketsII/LeetCode3477FruitsIntoBasketsII.go
package LeetCode3477FruitsIntoBasketsII

func numOfUnplacedFruits(fruits []int, baskets []int) int {

	fruitsPlaced := 0
	basketsLength := len(baskets)
	for _, fruit := range fruits {
		for i := 0; i < basketsLength; i++ {
			if fruit > baskets[i] {
				continue
			}
			baskets[i] = -1
			fruitsPlaced++
			break
		}
	}
	return len(fruits) - fruitsPlaced
}
